### ISSUE TITLE
replace call to functions from the string module in bin

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5019]
|<img alt="techtonik's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| techtonik reported|
|-|-|
|Trac ID|trac#5019|
|Type|enhancement|
|Created|2011-04-03 21:01:16Z|

For future py3k migration replace calls to `string` module with `str` object methods.

Attachments:

* [5019.bin.string.replace.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/ad1/ad1e57b13ad27a4e519a0b6e1e8afff294949679/93abe2b9481d182c95a4d80a82051a59e9fa41dd.patch) (6097 bytes) - added by techtonik on 2011-04-03 21:01:42Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5019 5019
type__enhancement enhancement
reporter__techtonik techtonik
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__py3k py3k
time__1301864476000000 1301864476000000
changetime__1301961756000000 1301961756000000
version__None None
owner__ 
cc__techtonik
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph commented|
|-|-|

(The "easy" keyword is for new contributors to find stuff to do; as this is a patch from a non-committer, it needs to be reviewed by a core committer in order to be added, so it can't be something for a new contributor to do.  Therefore, removing the keyword, even though it may indeed be easy.  Sorry for the confusion, this should be documented somewhere!)

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_3"></a>|@mwhudson commented|
|-|-|

(In [31543]) Branching to 'no-string-module-py3k-5019'

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_4"></a>|@mwhudson commented|
|-|-|

Thanks for the change, it is nice and simple.  I'll add a news file and merge it.

I notice in passing that (a) many "Twisted preambles" do not match the one in the coding standard and (b) there are still a lot of imports of the string module in Twisted, so if you're looking for something else to work on...

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_5"></a>|@mwhudson set status to `closed`|
|-|-|

(In [31546]) Merge no-string-module-py3k-5019: Remove imports of the string module from py3k.

Author: techtonik
Reviewer: mwh
Fixes: [#5019](https://github.com/twisted/twisted/issues/5019)

Replace string module calls with str object methods in bin/.
